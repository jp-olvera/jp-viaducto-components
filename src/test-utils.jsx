import React from 'react';
import { render } from '@testing-library/react';
import { ConfigProvider } from './providers';

const AllTheProviders = ({ children }) => {
  return <ConfigProvider>{children}</ConfigProvider>;
};

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
