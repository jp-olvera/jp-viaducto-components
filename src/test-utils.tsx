import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { ConfigProvider } from './providers';

const AllTheProviders = ({ children }: any) => (
  <ConfigProvider>{children}</ConfigProvider>
);

jest.mock('./cells/Input/amex.svg', () => null);
jest.mock('./cells/Input/mastercard.svg', () => null);
jest.mock('./cells/Input/visa.svg', () => null);
jest.mock('./cells/Dropdown/sorting.svg', () => null);
jest.mock('./cells/BareButton/close.svg', () => null);

const customRender = (ui, options?) => render(ui, { wrapper: AllTheProviders, ...options });
// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
