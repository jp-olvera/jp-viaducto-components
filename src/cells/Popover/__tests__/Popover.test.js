/* eslint-env jest */

import React, { useState, useRef } from 'react';
import '@testing-library/jest-dom/extend-expect';
import '@testing-library/react';
import { render, fireEvent } from '../../../test-utils';
import { Popover } from '..';

const clickOutside = jest.fn();
const Template = ({ args }) => {
  const [active, setActive] = useState(false);
  const handleClick = () => {
    setActive(!active);
  };
  const ref = useRef(null);
  return (
    <div>
      <button type='button'>outside</button>
      <button ref={ref} type='button' onClick={handleClick}>
        show
      </button>
      <Popover
        active={active}
        content={
          <span style={{ width: '300px', height: '300px' }}>content</span>
        }
        target={ref}
        handleClose={clickOutside}
        {...args}
      />
    </div>
  );
};
describe('<Popover/>', () => {
  test('should not render', () => {
    const { queryByText } = render(
      <Popover
        active
        content={
          <span style={{ width: '300px', height: '300px' }}>content</span>
        }
      />,
    );
    expect(queryByText('content')).toEqual(null);
  });
  test('should render properly', () => {
    const { container } = render(<Template />);
    expect(container).toMatchSnapshot();
  });
  test('should render elevation', () => {
    const { container } = render(
      <Template elevation={1} elevationDirection='bottom' />,
    );
    expect(container).toMatchSnapshot();
  });
  test('should render top', () => {
    const { container } = render(<Template position='top' />);
    expect(container).toMatchSnapshot();
  });
  test('should render bottom', () => {
    const { container } = render(<Template position='bottom' />);
    expect(container).toMatchSnapshot();
  });
  test('should render right', () => {
    const { container } = render(<Template position='right' />);
    expect(container).toMatchSnapshot();
  });
  test('should render left', () => {
    const { container } = render(<Template position='left' />);
    expect(container).toMatchSnapshot();
  });

  test('popover content should not be visible', () => {
    const { queryByText } = render(<Template />);
    expect(queryByText('content')).toEqual(null);
  });
  test('popover content should be visible', () => {
    const { getByText } = render(<Template />);
    fireEvent.click(getByText('show'));
    expect(getByText('content')).toBeInTheDocument();
  });

  test('popover should call onClose when mouseup outside', () => {
    const { getByText } = render(<Template position='right' />);
    fireEvent.click(getByText('show'));
    expect(getByText('content')).toBeInTheDocument();
    fireEvent.mouseUp(getByText('outside'));
    expect(clickOutside).toBeCalledTimes(1);
  });
  test('popover content is wider than window width', () => {
    const { getByText } = render(<Template />);
    fireEvent.click(getByText('show'));
    window.innerWidth = 299;
    window.innerHeight = 800;
    fireEvent(window, new Event('resize'));
    expect(getByText('content')).toBeVisible();
  });
  test('popover visible psition top but no space', () => {
    const { getByText } = render(<Template position='top' />);
    fireEvent.click(getByText('show'));
    window.innerWidth = 500;
    window.innerHeight = 800;
    fireEvent(window, new Event('resize'));
    expect(getByText('content')).toBeVisible();
  });
  test('content visible position right but no space', () => {
    const { getByText } = render(<Template position='right' />);
    fireEvent.click(getByText('show'));
    expect(getByText('content')).toBeVisible();
    window.innerWidth = 300;
    window.innerHeight = 800;
    fireEvent(window, new Event('resize'));
    expect(getByText('content')).toBeInTheDocument();
  });
  test('content visible position left but no space', () => {
    const { getByText } = render(<Template position='left' />);
    fireEvent.click(getByText('show'));
    expect(getByText('content')).toBeVisible();
    window.innerWidth = 300;
    window.innerHeight = 800;
    fireEvent(window, new Event('resize'));
    expect(getByText('content')).toBeInTheDocument();
  });
});
