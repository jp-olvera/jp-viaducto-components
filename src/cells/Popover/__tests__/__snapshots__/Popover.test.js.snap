// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Popover/> should render bottom 1`] = `
<div>
  <div>
    <button
      type="button"
    >
      outside
    </button>
    <button
      type="button"
    >
      show
    </button>
  </div>
</div>
`;

exports[`<Popover/> should render elevation 1`] = `
<div>
  <div>
    <button
      type="button"
    >
      outside
    </button>
    <button
      type="button"
    >
      show
    </button>
  </div>
</div>
`;

exports[`<Popover/> should render left 1`] = `
<div>
  <div>
    <button
      type="button"
    >
      outside
    </button>
    <button
      type="button"
    >
      show
    </button>
  </div>
</div>
`;

exports[`<Popover/> should render properly 1`] = `
<div>
  <div>
    <button
      type="button"
    >
      outside
    </button>
    <button
      type="button"
    >
      show
    </button>
  </div>
</div>
`;

exports[`<Popover/> should render right 1`] = `
<div>
  <div>
    <button
      type="button"
    >
      outside
    </button>
    <button
      type="button"
    >
      show
    </button>
  </div>
</div>
`;

exports[`<Popover/> should render top 1`] = `
<div>
  <div>
    <button
      type="button"
    >
      outside
    </button>
    <button
      type="button"
    >
      show
    </button>
  </div>
</div>
`;
